specs/protocol/superchain-config.md

```diff
@@ -5,60 +5,245 @@
+trusted party and is unlikely to exploit these conditions, this invariant is considered a High but
+not Critical impact condition.
+
+### iSUPC-003: The Guardian can revoke the Pause Deputy role at any time
+
+We require that the [Guardian](#guardian) be able to revoke the [Pause Deputy](#pause-deputy) role
+at any time. This condition is necessary to prevent a misbehaving Pause Deputy from continuing to
+trigger the [Pause Mechanism](#pause-mechanism) when this would not be desired by the Guardian
+itself.
+
+#### Impact
+
+**Severity: High**
+
+If this invariant were broken, assuming that [iSUPC-002][iSUPC-002] the Pause Deputy would
+potentially be able to repeatedly trigger the Pause Mechanism even if the Guardian does not want
+this to happen. We consider this to be a High severity condition because the Guardian can choose
+not to renew the pause to allow the system to operate if truly necessary.
+
+## Function Specification
+
+### initialize
+
+- MUST only be triggerable once.
+- MUST set the value of the Guardian role.
+- MUST set the value of the pause expiry period.
+
+### guardian
+
+Returns the address of the current [Guardian](#guardian).
+
+### pause
+
+Allows the [Guardian](#guardian) to trigger the [Pause Mechanism](#pause-mechanism). `pause` takes
+an address [Pause Identifier](#pause-identifier) as an input. This identifier determines which
+systems or chains are affected by the pause.
+
+- MUST revert if called by an address other than the Guardian.
+- MUST revert if the pausable flag for the given identifier is set to 1 (used/unavailable).
+- MUST set the pause timestamp for the given identifier to the current block timestamp.
+- MUST set the pausable flag for the given identifier to 1 (used/unavailable) to prevent repeated
+  pauses without a reset.
+
+### unpause
+
+Allows the [Guardian](#guardian) to explicitly unpause the system for a given
+[Pause Identifier](#pause-identifier) rather than waiting for the pause to expire. Unpausing a
+specific identifier does NOT unpause the global pause (zero address identifier). If the global
+pause is active, all systems will remain paused even if their specific identifiers are unpaused.
+
+- MUST revert if called by an address other than the Guardian.
+- MUST set the pause timestamp for the given identifier to 0, representing "not paused".
+- MUST set the pausable flag for the given identifier to 0 (ready to pause), allowing the pause
+  mechanism to be used again.
+
+### pausable
+
+Allows any user to check if the [Pause Mechanism](#pause-mechanism) can be triggered for a specific
+[Pause Identifier](#pause-identifier). The pausable status of a specific identifier is independent
+of the pausable status of the global pause (zero address identifier).
+
+- MUST return true if the pausable flag for the given identifier is 0 (ready to pause).
+- MUST return false if the pausable flag for the given identifier is 1 (used/unavailable).
+
+### paused
```

Comment:
Instead of handling the conditional for the global pause, I felt it would be better if these functions were simpler and just took a single pause identifier as a parameter. Then, inside of the `SystemConfig`, I handle checking for both the local and global pause identifiers.